tests/functional/m/modified_iterating.py

```diff
@@ -0,0 +1,42 @@
+"""Tests for iterating-modified messages"""
+# pylint: disable=not-callable,unnecessary-comprehension
+
+import copy
+
+item_list = [1, 2, 3]
+for item in item_list:  # [modified-iterating-list]
+    item_list.append(item)
+
+for item in item_list:  # [modified-iterating-list]
+    item_list.remove(item)
+    item_list.append(1)
+
+for item in item_list.copy():
+    item_list.append(item)
+    item_list.remove(item)
+
+for item in copy(item_list):
+    item_list.append(item)
+    item_list.remove(item)
+
+for item in [k for k in item_list]:
+    item_list.append(item)
+    item_list.remove(item)
+
+my_dict = {"1": 1, "2": 2, "3": 3}
+i = 1
+for item in my_dict:      # [modified-iterating-dict]
+    my_dict[i] = 1
+    i += 1
+
+i = 1
+for item in my_dict.copy():
+    my_dict[i] = 1
+    i += 1
+
+item_set = {1, 2, 3}
+for item in item_set:  # [modified-iterating-set]
+    item_set.add(item + 10)
+
+for item in item_set.copy():
+    item_set.add(item + 10)
```

Comment:
Could you add some tests with nested for loops?

And if you can think of one, a test where two types are modified like you mentioned in your last comment.

Reply:
https://github.com/PyCQA/pylint/pull/5628/files/7be0f88cf01f91a0dc72c18e77d9ab65c4301341#r790330656

Reply:
I'm not sure I understand what you're referring to. That isn't a nested for loop, I meant:
```python
for x in y:
    for a in z:
        pass
```

It also doesn't change two types?

Reply:
added